### DISCUSSION TITLE
Add default display name to auto create alias rules

### DISCUSSION THREAD

**Discussion by fossrob** [simple-login/app#821] [Feature Requests]
Hi 👋🏻 

Just discovered this today, and within a few hours of the trial I'd bought a Premium subscription! This is incredible! ❤️ 

The feature request I'd like to propose is to extend the _Domain Auto Create Rule_ functionality to have a _Display Name_ per rule. I've set up regex prefixes for myself, and my wife, but the _Display Name_ when replying to incoming mail defaults to 'blank'.

This would save having to go back and add the _Display Name_ to each auto created alias after it's created.

Just a thought! ❤️

**Comment by willymacoy** [comment]
Yes, please. This.

**Comment by dinvlad** [comment]
Any updates on this? (sorry to bump)

**Comment by vbc4ejyg** [comment]
Disappointed this request is almost 4-years old and still not implemented.  Exact same problem, and the regex ability is brilliant, but then falls flat in automation because you can't set the default Display Name.  It's a core part of basic email, especially to avoid getting it marked as spam.  Hopefully someone can make some updates soon.

SimpleLogin deserves Simple Rules.  Auto Create Alias Rules need to have a box to specify the Default Display Name for each match.

**Comment by vbc4ejyg** [comment]
For those still looking for a solution, addy.io does this perfectly with the "rules" feature. For any domain, you can set a rule that if any reply comes from your true mailbox address ([realaddress@proton.me](mailto:realaddress@proton.me) for example), it adds whatever "From Name" you want to your alias when your recipient sees. Pure genius implementation, and a great solution that can be implemented on any regex match you want!

I've been evaluating both simplelogin and addy.io, and not only does this solve this basic alias management requirement for new aliases, addy also is getting regular updates including a modern Android app. SimpleLogin could easily fix this as shown in the image below.

Simple fix:

<img width="934" height="1195" alt="image" src="https://github.com/user-attachments/assets/17e1638d-3a60-4b70-9043-deb575e21d85" />

**Comment by nguyenkims** [comment]
It's available now.

**Comment by vbc4ejyg** [comment]
This is absolutely fantastic, and thank you very much for a quick response @nguyenkims!! I have landed with SimpleLogin so this takes a lot of work out of the auto rule create workflow for me!  I have tested it and it works as expected, so I believe you can close this one now.

**Comment by shmatt** [comment]
A further suggestion. Could you allow the display name to use regex substitutions?

For example, `john.smith@domain.com` with the regex `^([a-zA-Z]+)\.([a-zA-Z]+)$` and the display name of `$1 $2` would produce a display name of `john smith`

or even better, the regex `^([a-zA-Z])([a-zA-Z]+)\.([a-z])([a-zA-Z]+)$` and display name `\U$1\E$2 \U$3\E$4` would produce `John Smith`
